sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/state/StateStoreSuite.scala

```diff
@@ -876,12 +889,17 @@ abstract class StateStoreSuiteBase[ProviderClass <: StateStoreProvider]
       assert(getLatestData(provider) === Set(("b", 0) -> 2))
 
       // Trying to get newer versions should fail
-      intercept[Exception] {
+      var e = intercept[SparkException] {
         provider.getStore(2)
       }
-      intercept[Exception] {
+      assert(e.getCause.isInstanceOf[SparkException])
+      assert(e.getCause.getMessage.contains("does not exist"))
```

Comment:
Use `checkError`

Reply:
I'm also not sure how to check this error because CANNOT_READ_DELTA_FILE is thrown through StateStoreSuite and CANNOT_READ_STREAMING_STATE_FILE is thrown through RocksDBStateStoreSuite

Reply:
What if parameters are different for these error classes? I guess we are generally no longer be able to use checkError if it can conditionally match to multiple error classes.